src/Model/Behavior/HashidBehavior.php

```diff
@@ -225,17 +226,21 @@ public function findHashed(Query $query, array $options) {
 		$idField = $this->_primaryKey;
 
 		$query->formatResults(function ($results) use ($field, $idField) {
-			return $results->map(function ($row) use ($field, $idField) {
-				if (empty($row[$idField])) {
-					return $row;
+			/** @var Collection $results */
+			$newResult = [];
+			$results->each(function ($row, $key) use ($field, $idField, &$newResult) {
+				if (!empty($row[$idField])) {
+					$row[$field] = $this->encodeId($row[$idField]);
+					if ($row instanceof Entity) {
+						$row->dirty($field, false);
+					}
+					$newResult[] = $row;
+				} elseif (is_string($row)) {
+					$newResult[$this->encodeId($key)] = $row;
 				}
-
-				$row[$field] = $this->encodeId($row[$idField]);
-				if ($row instanceof Entity) {
-					$row->dirty($field, false);
-				}
-				return $row;
 			});
+
+			return new Collection($newResult);
```

Comment:
because I made an array to wrap the new results and the `uery->formatResults` must return a Collection.
If you ask also why to use an array instead of making a temp array, the reason is that I couldnt fix the ids immediately to collection